float32 translation_error
float32 rotation_error
